File: Backend/mapping/src/main/java/com/mhp/coding/challenges/mapping/models/db/Image.java
package com.mhp.coding.challenges.mapping.models.db;

import com.mhp.coding.challenges.mapping.models.dto.ImageDto;
import java.util.ArrayList;

public class Image extends DBEntity {

    private String url;

    private ImageSize imageSize;

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public ImageSize getImageSize() {
        return imageSize;
    }

    public void setImageSize(ImageSize imageSize) {
        this.imageSize = imageSize;
    }
    
    public ImageDto map() {
        ImageDto imageDto = new ImageDto();
        
        imageDto.setId(getId());
        imageDto.setUrl(getUrl());
        imageDto.setImageSize(getImageSize());
        
        return imageDto;
    }    
}

File: Backend/mapping-kotlin/src/main/kotlin/com/mhp/coding/challenges/mapping/models/db/Image.kt
package com.mhp.coding.challenges.mapping.models.db

import com.mhp.coding.challenges.mapping.models.dto.ImageDto

import java.util.*

class Image(
    var url: String,
    var imageSize: ImageSize,
    override var id: Long,
    override var lastModified: Date,
    override var lastModifiedBy: String? = null
) : DBEntity {
    
    fun map() : ImageDto {
        val imageDto = ImageDto(
            id = this.id,
            url = this.url,
            imageSize = this.imageSize
        )
        return imageDto
    }
}

enum class ImageSize {
    SMALL,
    MEDIUM,
    LARGE,
}
